lib/strict_ivars/base_processor.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "securerandom"
+
+class StrictIvars::BaseProcessor < Prism::Visitor
+	EVAL_METHODS = Set[:class_eval, :module_eval, :instance_eval, :eval].freeze
+
+	#: (String) -> String
+	def self.call(source)
+		visitor = new
+		visitor.visit(Prism.parse(source).value)
+		buffer = source.dup
+		annotations = visitor.annotations
+		annotations.sort_by!(&:first)
+
+		annotations.reverse_each do |offset, string|
+			buffer.insert(offset, string)
+		end
+
+		buffer
+	end
+
+	def initialize
+		@context = Set[]
+		@annotations = []
+	end
+
+	#: Array[[Integer, :start_ivar_read | :end_ivar_read, Symbol]]
+	attr_reader :annotations
+
+	def visit_call_node(node)
+		name = node.name
+
+		if EVAL_METHODS.include?(name) && (arguments = node.arguments)
+			location = arguments.location
+
+			if node.receiver
+				receiver_local = "__eval_receiver_#{SecureRandom.hex(8)}__"
```

Comment:
We need to store the specific receiver with a unique name so we can identify it later. This enables us to check if the receiver of this argument has had the eval method overridden.